### PR TITLE
WIP: Initial code for implementing tree shaking using rollup loader

### PR DESCRIPTION
**Pull request by sebworks**
## Changes

- Modified  `config/webpack-config.js` to add the `webpack-rollup` loader and prevent babel from [transpiling](https://github.com/erikdesjardins/webpack-rollup-loader#usage) the modules. This is necessary in order to properly implement tree shaking which doesn't work properly when using UglifyJS and Webpack.  

## Testing

1. Run `npm install`
2. Run gulp clean && gulp build.
3. Notice that the VideoPlayer module is smaller. 

## Screenshots

Before: 

screen shot 2017-11-30 at 7 05 28 am

After:

screen shot 2017-11-30 at 10 49 19 pm

## Notes

- I've only converted code which uses the `error-messages-config` file. You will see additional benefits when the rest of the code is converted.

- You might notice small increases or decreases in some bundles due to rollup parsing and dead code elimination.

## Todos

-

## Checklist

* [ ] PR has an informative and human-readable title
* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
* [ ] Passes all existing automated tests
* [ ] Any *change* in functionality is tested
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated
* [ ] Reviewers requested with the [Reviewer tool](https://help.github.com/articles/about-pull-request-reviews/) :arrow_right:

### CODE REVIEW

File: cfgov/unprocessed/js/config/error-messages-config.js [hunk 1]
```diff
@@ -6,47 +6,53 @@
...
-    ONE_REQURED_ES: 'Por favor, ingrese como mínimo una fecha.'
-  },
-  EMAIL: {
+const COMMENT = {
+  REQUIRED: 'Please enter a comment.',
+  REQUIRED_ES: 'Por favor, introduzca un comentario.'
+}
+
+const CHECKBOX = {
+  REQUIRED: 'Please select at least %s o†f the options.'
```
Comment (anselmbradford) [inline cfgov/unprocessed/js/config/error-messages-config.js]: `o†f` correct?

File: config/webpack-config.js [hunk 2]
```diff
@@ -17,17 +17,27 @@ const COMMON_BUNDLE_NAME = 'common.js';
...
           },
           debug: true
         } ] ]
       }
-    } ],
+    } ,
+    exclude: {
+      test: /node_modules/,
+      // The below regex will capture all node modules that start with `cf`
+      // or atomic-component. Regex test: https://regex101.com/r/zizz3V/1/.
```
Comment (anselmbradford) [inline config/webpack-config.js]: Do we need `atomic-component`? It's `cf-atomic-component` in CF.

Comment (anselmbradford) [reply config/webpack-config.js]: 👍  to having the regex test link.

File: config/webpack-config.js [hunk 3]
```diff
@@ -17,17 +17,27 @@ const COMMON_BUNDLE_NAME = 'common.js';
...
+      exclude: /node_modules\/(?:cf.+|atomic-component)/
+    }
+  }, {
+    test: /\.js$/,
+    loader: 'webpack-rollup-loader',
```
Comment (anselmbradford) [inline config/webpack-config.js]: https://github.com/erikdesjardins/webpack-rollup-loader#usage shows webpack-rollup-loader appearing before babel. Here's it's after. Does that make a difference?

File: cfgov/unprocessed/js/config/error-messages-config.js [hunk 4]
```diff
@@ -6,47 +6,53 @@
...
+};
+
+const DEFAULT = 'Error!';
+
+const DOM = {
+  INVALID: 'Invalid dom element was provided.'
+};
+
+const EMAIL = {
     INVALID: 'You have entered an invalid email address.',
```
Comment (anselmbradford) [inline cfgov/unprocessed/js/config/error-messages-config.js]: TRIVIAL: These four can be 2-space indent to match the others.

File: cfgov/unprocessed/js/config/error-messages-config.js [hunk 5]
```diff
@@ -6,47 +6,53 @@
...
+    SUCCESS: 'Your submission was successfully received.',
+    SUCCESS_ES: 'Su presentación fue recibido con éxito.'
   }
 };
 
-module.exports = Object.freeze( ERROR_MESSAGES );
+const OPTION = {
+  REQUIRED: 'Please select an option.',
+  REQUIRED_ES: 'Por favor, seleccione una opción.'
+}
```
Comment (anselmbradford) [inline cfgov/unprocessed/js/config/error-messages-config.js]: TRIVIAL: missing trailing ;

File: config/webpack-config.js [hunk 6]
```diff
@@ -17,17 +17,27 @@ const COMMON_BUNDLE_NAME = 'common.js';
...
         presets: [ [ 'env', {
+          modules: false,
           targets: {
             browsers: BROWSER_LIST.LAST_2_IE_9_UP
           },
           debug: true
         } ] ]
       }
-    } ],
+    } ,
```
Comment (anselmbradford) [inline config/webpack-config.js]: Should be `},` (no space)

File: config/webpack-config.js (hunk 3 above)
Comment (sebworks) [reply config/webpack-config.js]: That is a good question, I'll check on this.

**Comment by sebworks** [comment]
closing, [webpack 4](https://medium.com/webpack/webpack-4-beta-try-it-today-6b1d27d7d7e2) now has proper tree shaking and dead code elimination.

**Pull request closed without merging**